src/cli/list.rs

```diff
@@ -0,0 +1,283 @@
+        .set_content_arrangement(ContentArrangement::Dynamic);
+
+    // Add headers
+    table.set_header(vec![
+        Cell::new("Package").add_attribute(Attribute::Bold),
+        Cell::new("Version").add_attribute(Attribute::Bold),
+        Cell::new("Build").add_attribute(Attribute::Bold),
+        Cell::new("Size").add_attribute(Attribute::Bold),
+        Cell::new("Kind").add_attribute(Attribute::Bold),
+        Cell::new("Channel").add_attribute(Attribute::Bold),
+    ]);
+
+    for package in packages {
+        // Convert size to human readable format
+        let size_human = match package.size_bytes {
+            Some(size_bytes) => human_bytes(size_bytes as f64).to_string(),
+            None => "".to_string(),
+        };
+
+        let package_name = if package.is_explicit {
+            Cell::new(package.name)
+                .fg(Color::Green)
+                .add_attribute(Attribute::Bold)
+        } else {
+            Cell::new(package.name)
+        };
+
+        table.add_row(vec![
+            package_name,
+            Cell::new(package.version),
+            Cell::new(package.build.unwrap_or_else(|| "".to_string())),
+            Cell::new(size_human),
+            Cell::new(package.kind),
+            Cell::new(package.channel.unwrap_or_else(|| "".to_string())),
+        ]);
+    }
+
+    println!("{table}");
+}
+
+fn json_packages(packages: Vec<PackageToOutput>, json_pretty: bool) {
+    let json_string = if json_pretty {
+        serde_json::to_string_pretty(&packages)
+    } else {
+        serde_json::to_string(&packages)
+    }
+    .map_err(|_| miette::miette!("Cannot serialize packages to JSON"))
+    .unwrap();
+
+    println!("{}", json_string);
+}
+
+fn create_package_to_output(
+    p: &LockedDependency,
+    project_dependency_names: &[String],
+) -> PackageToOutput {
+    let name = p.name.clone();
+    let version = p.version.clone();
+
+    let kind = match p.kind {
+        LockedDependencyKind::Conda(_) => "conda".to_string(),
+        LockedDependencyKind::Pypi(_) => "pypi".to_string(),
+    };
+    let build = match p.kind {
+        LockedDependencyKind::Conda(_) => p.as_conda().unwrap().build.clone(),
+        LockedDependencyKind::Pypi(_) => p.as_pypi().unwrap().build.clone(),
+    };
+
+    let size_bytes = match p.kind {
+        LockedDependencyKind::Conda(_) => p.as_conda().unwrap().size,
+        LockedDependencyKind::Pypi(_) => None,
+    };
+
+    let channel = Some("".to_string());
+
+    // NOTE(hadim): does not work - returns `conda-forge/osx-arm64/tk-8.6.13-h5083fa2_1.conda`
+    // let channel = match p.kind {
+    //     LockedDependencyKind::Conda(_) => {
+    //         Channel::from_url(
+    //             p.as_conda().unwrap().url.clone(),
+    //             Some(vec![Platform::current()]),
+    //             &ChannelConfig::default(),
+    //         )
+    //         .name
+    //     }
+    //     LockedDependencyKind::Pypi(_) => Some("".to_string()),
```

Comment:
Maybe `channel` could be changed into `source`, as we are also adding a pypi_index option really soon. So you can use private `pypi` registries.